gapic/templates/testing/_default_constraints.j2

```diff
@@ -1,7 +1,15 @@
+{# TODO(https://github.com/googleapis/gapic-generator-python/issues/2121): Remove the following variable (and the condition later in this file) for async rest transport once support for it is GA. #}
+{% set rest_async_io_enabled = api.all_library_settings[api.naming.proto_package].python_settings.experimental_features.rest_async_io_enabled %}
 {% from '_pypi_packages.j2' import pypi_packages %}
 # This constraints file is required for unit tests.
 # List all library dependencies and extras in this file.
 google-api-core
+{% if rest_async_io_enabled %}
+google-auth
+# from google-auth[aiohttp]
+aiohttp
+requests
```

Comment:
We shouldn't include the `async_rest` extra in all constraints because we won't have any tests without the `async_rest` extra installed.

Reply:
addressed.

Reply:
addressed. See my other comment and discussing this over our offline chat.